apps/parser/controllers/index.js

```diff
@@ -95,3 +96,51 @@ export const updateAllPosts = async (req, res) => {
   }
   res.status(200).json(results);
 };
+
+const getOriginalFileName = (post, url) => {
+  let fileName = "";
+  if (url.includes("scontent")) {
+    fileName = `${post.id}/images/${url.split("?")[0].split("/").pop()}`;
+  } else if (url.includes("external")) {
+    fileName = `${post.id}/images/${url
+      .split("&url=")[1]
+      .split("&cfs=")[0]
+      .replace("%3A", ":")
+      .replace("%2F", "/")}`;
+  }
+  return fileName;
+};
+
+export const selfHostAllImages = async (_req, res) => {
+  const posts = (await fetchAllPosts()).chunk(20);
+  let updatedPosts = [];
+  let updatedImages = [];
+  const results = [];
+  let newUrl;
+  let img;
+  for (let i = 0; i < posts.length; i++) {
+    for (let j = 0; j < posts[i].length; j++) {
+      if (
+        posts[i][j]._doc.images.length &&
+        posts[i][j]._doc.images[0].includes("uwcribs")
+      )
+        continue;
+      for (let k = 0; k < posts[i][j]._doc.images.length; k++) {
+        img = posts[i][j]._doc.images[k];
+        if (!img.includes("scontent") || !img.includes("external")) continue;
```

Comment:
is this hard coded from the squidward image? What if we just host the squidward image ourselves in the bucket

Reply:
This was just to check whether the images are from FB, if they aren't the image is skipped.